daemon/k8s_watcher.go

```diff
@@ -269,11 +270,31 @@ func blockWaitGroupToSyncResources(waitGroup *sync.WaitGroup, informer cache.Con
 			scopedLog.Fatalf("failed to wait for cache to sync")
 		}
 		scopedLog.Debug("cache synced")
-		waitGroup.Done()
+		d.k8sResourceSyncedMu.RLock()
+		c := d.k8sResourceSynced[resourceName]
+		d.k8sResourceSyncedMu.RUnlock()
+		close(c)
```

Comment:
I think this close should be done before the lock is released, and then after the close, you should do `delete(d.k8sResourceSynced, resourceName)` because otherwise the same channel could be accessed and attempted to be closed by different goroutines depending on how this structure (`d.k8sResourceSynced`) is utilized in the future. 
Or, you can do : 

```
select {
case <-c:
default:
    close(c) 
}
```

As a means of checking if the channel has been closed already.

Reply:
Good find